geode/dispatcher.py

```diff
@@ -254,7 +262,10 @@ def __init__(self, config=None):
         else:
             merged_df = pd.concat([estimate_df, res_df], sort=False)
 
-        merged_df = merged_df[~merged_df.index.duplicated(keep='last')].sort_index()
+        merged_df = merged_df[~merged_df.index.duplicated(keep='last')].reindex(index=idx.index, copy=False)
+
+        if return_inverse:
+            return merged_df, inv
 
         return merged_df
```

Comment:
Above we're returning a tuple. Do we need to do so as well here?

Reply:
return_inverse is a special use case, will change signature